frontends/PyRTG/test/lit.cfg.py

```diff
@@ -0,0 +1,79 @@
+# -*- Python -*-
+
+import os
+import platform
+import re
+import shutil
+import subprocess
+import tempfile
+import warnings
+
+import lit.formats
+import lit.util
+
+from lit.llvm import llvm_config
+from lit.llvm.subst import ToolSubst
+from lit.llvm.subst import FindTool
+
+# Configuration file for the 'lit' test runner.
+
+# name: The name of this test suite.
+config.name = 'PyRTG'
+
+config.test_format = lit.formats.ShTest(not llvm_config.use_lit_shell)
+
+# suffixes: A list of file extensions to treat as test files.
+config.suffixes = ['.mlir', '.py']
+
+# test_source_root: The root path where tests are located.
+config.test_source_root = os.path.dirname(__file__)
+
+# test_exec_root: The root path where tests should be run.
+config.test_exec_root = os.path.join(config.circt_obj_root, 'test')
+
+config.substitutions.append(('%PATH%', config.environment['PATH']))
+config.substitutions.append(('%shlibext', config.llvm_shlib_ext))
+config.substitutions.append(('%shlibdir', config.circt_shlib_dir))
+config.substitutions.append(('%INC%', config.circt_include_dir))
+config.substitutions.append(('%PYTHON%', f'"{config.python_executable}"'))
+
+llvm_config.with_system_environment(['HOME', 'INCLUDE', 'LIB', 'TMP', 'TEMP'])
+
+llvm_config.use_default_substitutions()
+
+# Set the timeout, if requested.
+if config.timeout is not None and config.timeout != "":
+  lit_config.maxIndividualTestTime = int(config.timeout)
+
+# excludes: A list of directories to exclude from the testsuite. The 'Inputs'
+# subdirectories contain auxiliary inputs for various tests in their parent
+# directories.
+config.excludes = [
+    'CMakeLists.txt', 'README.txt', 'LICENSE.txt', 'lit.cfg.py',
+    'lit.local.cfg.py'
+]
+
+# test_source_root: The root path where tests are located.
+config.test_source_root = os.path.dirname(__file__)
+
+# test_exec_root: The root path where tests should be run.
+config.test_exec_root = os.path.join(config.circt_obj_root,
+                                     'frontends/PyRTG/test')
+
+# Tweak the PATH to include the tools dir.
+llvm_config.with_environment('PATH', config.llvm_tools_dir, append_path=True)
+
+tool_dirs = [
+    config.circt_tools_dir, config.mlir_tools_dir, config.llvm_tools_dir
+]
+tools = ['py-split-input-file.py']
+
+llvm_config.add_tool_substitutions(tools, tool_dirs)
+
+# Tweak the PYTHONPATH to include the binary dir.
+llvm_config.with_environment(
+    'PYTHONPATH', [os.path.join(config.circt_python_packages_dir, 'pyrtg')],
+    append_path=True)
+
+# rtgtool shortcut
+config.substitutions.append(('%rtgtool%', f'{config.python_executable} {os.path.join(config.circt_python_packages_dir, 'pyrtg', 'rtgtool', 'rtgtool.py')}')) 
```

Comment:
Could the yapf issue be related to the nested `'` in the format string? It's a bit ugly, but maybe splitting the format string out into variables and interpolating those could work, or using a `... + " " + ...` instead to pull the complex interpolations out into expressions outside the string literal.

Reply:
Good call! Interesting that yapf just crashes instead of printing a useful error message in such a case.

Reply:
Yeah very strange indeed 🤔